logstash-core/src/main/java/org/logstash/ackedqueue/ext/JrubyAckedQueueExtLibrary.java

```diff
@@ -69,20 +67,15 @@ public IRubyObject ruby_initialize(ThreadContext context, IRubyObject[] args)
             int checkpointMaxInterval = RubyFixnum.num2int(args[5]);
             long queueMaxBytes = RubyFixnum.num2long(args[6]);
 
-            Settings s = new FileSettings(args[0].asJavaString());
             PageIOFactory pageIOFactory = (pageNum, size, path) -> new MmapPageIO(pageNum, size, path);
             CheckpointIOFactory checkpointIOFactory = (source) -> new FileCheckpointIO(source);
-            s.setCapacity(capacity);
-            s.setMaxUnread(maxUnread);
-            s.setQueueMaxBytes(queueMaxBytes);
-            s.setCheckpointMaxAcks(checkpointMaxAcks);
-            s.setCheckpointMaxWrites(checkpointMaxWrites);
-            s.setCheckpointMaxInterval(checkpointMaxInterval);
-            s.setElementIOFactory(pageIOFactory);
-            s.setCheckpointIOFactory(checkpointIOFactory);
-            s.setElementClass(Event.class);
-
-            this.queue = new Queue(s);
+            this.queue = new Queue(SettingsImpl.fileSettingsBuilder(args[0].asJavaString()).capacity(capacity)
+                .maxUnread(maxUnread).queueMaxBytes(queueMaxBytes)
+                .checkpointMaxAcks(checkpointMaxAcks)
+                .checkpointMaxWrites(checkpointMaxWrites)
+                .checkpointMaxInterval(checkpointMaxInterval)
+                .elementIOFactory(pageIOFactory).checkpointIOFactory(checkpointIOFactory)
+                .elementClass(Event.class).build());
```

Comment:
@robbavey if you ask me, no separate variable is kind of pointless. If you only use it once, it should probably only be there once.
That said ... horrible indentation here makes it very hard to read too, sorry about that, fixing :)